File: doctool/test/hostrecv.pl
#!/usr/bin/env perl
# name service test.
# 2018-12-18
use strict;
use warnings FATAL => 'all';
use Net::DNS;
use Data::Dumper;

unless (defined $ARGV[0]) {
  print "usage: $0 domainname\n";
  exit 1;
}

while (1) {
  my $res = Net::DNS::Resolver->new()
            or die "dns error:$!";

  my $a_query = $res->search($ARGV[0]);
  print "Website IP Address...\n";

  if ($a_query) {
    foreach my $rr ($a_query->answer) {
      next unless $rr->type eq "A";
      print $rr->address, "\n";
    }
  } else {
    warn "Unable to obtain A record: ", $res->errorstring, "\n";
  }
  sleep 2;
}

File: doctool/test/hostrecv.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import dns.resolver
import time
import sys

if len(sys.argv) == 1:
  print "usage: %s domainname" % sys.argv[0]
  sys.exit(1)

while 1:
  domain = sys.argv[1]
  A = dns.resolver.query(domain,'A')
  for i in A.response.answer:
     for j in i.items:
        print(j.address)
  time.sleep(2)
